src/NAR/Parser.java

```diff
@@ -40,4 +51,50 @@ public static boolean checkUserInput(String statement)
 		Editor.printToConsole("Unknown construct '" + calledConstructName + "' found.");
 		return false;
 	}
+	
+	/**
+	 * This method takes a variable name as input.
+	 * Returns true if the variable has been defined false otherwise.
+	 * @param variable
+	 * @return
+	 */
+	public static boolean isVarDefined(String variable) {
+		return Instance.variables.containsKey(variable);
+	}
+	
+	/**
+	 * This method takes two parameters, a variable name, and it�fs associated number value. 
+	 * It stores the variable name and number together in internal memory. 
+	 * It also can be used to store a new value for an already defined variable.
+	 * @param variable
+	 * @param value
+	 */
+	public static void defineVar(String variable, double value) {
+		Instance.variables.put(variable, value);
+	}
+	
+	/**
+	 * This method takes a variable name as input and 
+	 * returns the numeric value associated with it.
+	 * @param variable
+	 * @return value of variable
+	 */
+	public static double getVarValue(String variable) {
+		try {
+			return Instance.variables.get(variable);
```

Comment:
You could just let the program crash here because this function should never get called on a variable which is not defined. if that ever happens that's on us